trinity/sync/full/service.py

```diff
@@ -47,6 +47,7 @@ def __init__(self,
                 self.chain, self.chaindb, self.peer_pool, self.cancel_token)
             await chain_syncer.run()
 
+        await self.sleep(0)
```

Comment:
I don't think this will have the effect you want. This code is inside `_run()` and `BaseService.is_running` will return True as long as `_run()` is executing. I think the correct thing to do here is to check if the token has been triggered instead of checking for `self.is_running`